contrib/inkscape_tools/src/svg_to_png_and_nit.nit

```diff
@@ -14,51 +14,140 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# This script extracts pngs from a single svg for all objects with ids
-# beginning by 0. Requires Inkscape.
+# Extract images of objects from an SVG file using Inkscape
 module svg_to_png_and_nit
 
 import opts
 import template
 
+# Image info extracted from the SVG file
 class Image
+	# Name extracted from the object ID minus the `0` prefix and Nit safe
 	var name: String
+
+	# Left border
 	var x: Int
+
+	# Top border
 	var y: Int
+
+	# Image width
 	var w: Int
+
+	# Image height
 	var h: Int
+
+	# Right border
 	fun right: Int do return x+w
+
+	# Bottom border
 	fun bottom: Int do return y+h
 
 	redef fun to_s do return name
 end
 
-# The Nit source file to retreive all images
-class ImageSetSrc
+# Document being processed, concerns both the source and the target
+class Document
+	# Name of the source file
+	var drawing_name: String
+
+	# Name of the class to generate
+	var nit_class_name: String = drawing_name.capitalized + "Images" is lazy
+
+	# Scaling to apply to the exported image
+	var scale: Float
+
+	# Source minimum X
+	var min_x: Int
+
+	# Source maximum X
+	var max_x: Int
+
+	# Source minimum Y
+	var min_y: Int
+
+	# Source maximum Y
+	var max_y: Int
+
+	# Get the coordinates for `image` as `"x, y, w, h"`
+	fun coordinates(image: Image): String
+	do
+		var x = image.x.adapt(min_x, scale)
+		var y = image.y.adapt(min_y, scale)
+		var w = (image.w.to_f*scale).to_i
+		var h = (image.h.to_f*scale).to_i
+
+		return "{x}, {y}, {w}, {h}"
+	end
+end
+
+# Nit module with a single class to retrieve to access the extracted images
+abstract class ImageSetSrc
 	super Template
 
-	var name: String
-	init(name: String) do self.name = name
+	# Target document
+	var document: Document
 
-	var attributes = new Array[String]
-	var load_exprs = new Array[String]
+	# Images found in the source document
+	var images: Array[Image]
+end
+
+# Nit module targeting the MNit framework
+class MnitImageSetSrc
+	super ImageSetSrc
 
 	redef fun rendering
 	do
+		# Known array of images
+		var arrays_of_images = new Array[String]
+
+		# Attributes of the class
+		var attributes = new Array[String]
+
+		# Statements for the `load_all_` method
```

Comment:
The only `load_all_` method I can see if in fact named `load_all` without the last underscore. Are talking about other methods?